docs/0004-we-use-folder-convention-for-infrastructure-resource-definitions.md

```diff
@@ -0,0 +1,55 @@
+# 1. We use folder convention for infrastructure resource definitions
+
+Date: 2024-04-10
+
+## Status
+
+Accepted
+
+## Context
+
+Each monorepo holds the definition for infrastructure resources needed.
+
+## Decision
+
+The resource definitions will be placed into the `infra/resources` folder.
+Definitions are intended to work for an environment in a specific region. Each pair environment/region is a Terraform project on its own and they will be located in the `<env>/<region>` subfolder.
+
+Every automation will expect resources to be in such folders.
+
+#### Example
+
+```
+infra/
```

Comment:
why not directly under 'infra'?

Reply:
To keep 'real' resources (`prod/dev/uat` and `_modules` folders) separated from accessories such as identity and repo settings

Reply:
1. are we going to use `infra/identity` for identities?
2. what kind of repo settings do we plan to put under `infra`?

Reply:
I think we will use `infra/resources`, `infra/identity`, `infra/repository`

Reply:
2. repository data (name, description, tags), branch protection, branch deployment rules, environment federated with Azure, repository secrets and variables.

I'm facing some elasticity issues while implementing that module, we will discuss them in their PR very soon

Reply:
See https://github.com/pagopa/dx-typescript/pull/11/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5, there is a bit fo explanation